src/Adapter/ContainerBuilder.php

```diff
@@ -47,26 +91,183 @@ class ContainerBuilder
      */
     public static function getContainer($name, $isDebug)
     {
-        $containerName = ucfirst($name) . 'Container';
-        $file = _PS_CACHE_DIR_ . "${containerName}.php";
+        if (null === self::$container) {
+            $builder = new ContainerBuilder();
+            self::$container = $builder->initContainer($name, $isDebug);
+        }
 
-        if (!$isDebug && file_exists($file)) {
-            require_once $file;
+        return self::$container;
+    }
 
-            return new $containerName();
+    /**
+     * @param string $name
+     * @param bool $isDebug
+     *
+     * @return ContainerInterface|SfContainerBuilder|null
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function initContainer($name, $isDebug)
+    {
+        $this->containerName = $name;
+        $this->isDebug = $isDebug;
+
+        $this->environment = $this->getEnvironment();
+        $this->containerClassName = ucfirst($this->containerName) . 'Container';
+        $this->dumpFile = _PS_CACHE_DIR_ . $this->containerClassName . '.php';
+        $this->containerConfigCache = new ConfigCache($this->dumpFile, $this->isDebug);
+
+        //Necessary to require all annotation classes from Doctrine
+        Setup::createAnnotationMetadataConfiguration([]);
```

Comment:
```diff
- //Necessary to require all annotation classes from Doctrine
- Setup::createAnnotationMetadataConfiguration([]);
+ $this->loadDoctrineAnnotationsMetadata();
```

Reply:
This is actually use to include Doctrine classes (from `Doctrine\ORM\Configuration::newDefaultAnnotationDriver`):

```php
AnnotationRegistry::registerFile(__DIR__ . '/Mapping/Driver/DoctrineAnnotations.php');
```

Do you suggest I create a `loadDoctrineAnnotationsMetadata` to do this?

Reply:
@mickaelandrieu last comment to manage, but I'm still not sure what you mean?

Reply:
extract a non-meaningful behavior to its own function with a meaningful name instead of writing a comment :+1:

Reply:
Ok done